scripts/ipfs/install

```diff
@@ -23,8 +23,8 @@ rm -rf "$BASE_DIR/tmp"
 ipfs init || true
 
 # Configure HTTP to IPFS gateway
-# shellcheck source=../shared/nginx/install
-source "$BASE_DIR/../shared/nginx/install"
```

Comment:
I don't understand why it is moved out of `shared`. The fact that it is installed by default doesn't change the fact that it's shared?

This also makes me think whether it makes sense to start sorting modules into `default` vs. `optional`, since every module folder so far is an optional component. Now `nginx` is default, like `cjdns`, which never had its own module and kind of live within the main install scripts. My concern is, the knowledge of which ones are default and which are not currently is not expressed in code, making this repo hard to maintain.

Reply:
I did it originally because @darkdrgn2k suggested it in #248. I agree that default and optional would be good categories to separate things into, although I don't really like the idea of `nginx` (and `cjdns`) not having their own module. At the moment there is only one module in `shared`, `nodeinfo`, so I would be in favour of moving that out as well. What if I do that, and then I can make a separate PR for the `default` and `optional` folders?

Reply:
Nginx was previous installed only when needed (ipfs, patchfoo etc)
We have decided in that issue to make it a core because of nodeinfo.json and welcome page which we want to exist regardless of what is installed.

Therefor they now are part of the main script not a "default" install but a "required" install.

as for  CJDNS not being a module it is currently  tied in pretty hard into things (ie Access Point name and Host name and such) that would require more work to untangle.

Reply:
So which are we doing?
- nginx in `shared`
- classify into `default` vs. `optional`
- Leave nginx in root like cjdns

Reply:
moved conversation to https://github.com/tomeshnet/prototype-cjdns-pi/issues/248